frontend/controller/actions/identity.js

```diff
@@ -701,5 +701,72 @@ export default (sbp('sbp/selectors/register', {
   ...encryptedAction('gi.actions/identity/joinDirectMessage', L('Failed to join a direct message.')),
   ...encryptedAction('gi.actions/identity/joinGroup', L('Failed to join a group.')),
   ...encryptedAction('gi.actions/identity/leaveGroup', L('Failed to leave a group.')),
-  ...encryptedAction('gi.actions/identity/setDirectMessageVisibility', L('Failed to set direct message visibility.'))
+  ...encryptedAction('gi.actions/identity/setDirectMessageVisibility', L('Failed to set direct message visibility.')),
+  'gi.actions/identity/uploadFiles': async ({ attachments, billableContractID }: {
+    attachments: Object, billableContractID: string
+  }) => {
+    const rootGetters = sbp('state/vuex/getters')
+
+    try {
+      const attachmentsData = await Promise.all(attachments.map(async (attachment) => {
+        const { mimeType, url } = attachment
+        // url here is an instance of URL.createObjectURL(), which needs to be converted to a 'Blob'
+        const attachmentBlob = await objectURLtoBlob(url)
+        const response = await sbp('chelonia/fileUpload', attachmentBlob, {
+          type: mimeType, cipher: 'aes256gcm'
+        }, { billableContractID })
+        const { delete: token, download: downloadData } = response
+        return {
+          downloadData: { ...omit(attachment, ['url']), downloadData },
+          deleteData: { token }
+        }
+      }))
+
+      const tokensByManifestCid = attachmentsData.map(({ downloadData, deleteData }) => ({
+        manifestCid: downloadData.downloadData.manifestCid,
+        token: deleteData.token
+      }))
+
+      await sbp('gi.actions/identity/saveFileDeleteToken', {
+        contractID: rootGetters.ourIdentityContractId,
+        data: { tokensByManifestCid }
+      })
+
+      return attachmentsData.map(({ downloadData }) => downloadData)
+    } catch (err) {
+      console.error('Error during uploading files', err)
+    }
+  },
+  'gi.actions/identity/removeFiles': async ({ manifestCids, identityContractID }: {
+    manifestCids: string[], identityContractID: string, billableContractID?: string
+  }) => {
+    const rootGetters = sbp('state/vuex/getters')
+    const me = sbp('state/vuex/state').loggedIn.identityContractID
+
+    if (me !== identityContractID) {
+      return
+    }
+
+    try {
+      const tokensMap = {}
+      for (const manifestCid of manifestCids) {
+        const token = rootGetters.currentIdentityState.fileDeleteTokens[manifestCid]
+        if (!token) {
+          console.error(`Missing delete token for file with manifestCid ${manifestCid}`)
+          return
+        }
+        tokensMap[manifestCid] = { token }
+      }
+
+      await sbp('chelonia/fileDelete', manifestCids, tokensMap)
+      await sbp('gi.actions/identity/removeFileDeleteToken', {
+        contractID: identityContractID,
+        data: { manifestCids }
+      })
+    } catch (err) {
+      console.error(err?.message || err)
```

Comment:
"Naked" calls to `console.error/warn` logs are generally something we want to avoid.

In other words, we always want to include a string as the first parameter that describes where the error is coming from (and possibly why). Without this, it will be less clear in the console logs where the error is coming from.

So something like this would be preferable:

```js
console.error("'gi.actions/identity/removeFiles' error:", err?.message || err)
```

Also, all of these actions must behave consistently: this means throwing an exception when the error occurs. Both `'gi.actions/identity/uploadFiles'` and `'gi.actions/identity/removeFiles'` fail to do this.

Please look at how the other actions behave, using `GIErrorUIRuntimeError` to create a localized user-facing string that can be shown to the user if an error occurs.